### PR TITLE
Indent code in continue mode

### PR DESCRIPTION
**Pull request by k0kubun**
I implemented a feature to indent continuing code using text/scanner.
It increases the indent depth by '{' token, and decreases it by '}' token.

### CODE REVIEW

**Comment by motemen** [comment]
Thanks, mostly looks good, but I think the code which resets the line does not work on Windows (though I did not check it). Can we address this problem?

**Comment by k0kubun** [comment]
> I think the code which resets the line does not work on Windows (though I did not check it). Can we address this problem?

I ported the code to reset the line for windows.  e42d270
It works fine.

File: liner.go [hunk 1]
```diff
@@ -54,3 +63,55 @@ func (cl *contLiner) Accepted() {
...
+	cursorUp()
+	if len(lines) > 1 {
+		fmt.Printf("\r%s", promptContinue)
+	} else {
+		fmt.Printf("\r%s", promptDefault)
+	}
+
+	cl.printIndent(min)
+	fmt.Print(lastLine)
+	cl.printIndent(max - min)
```
Comment (motemen) [inline liner.go]: What is line is for? I guess erasing previous input by overwriting with spaces?

Comment (k0kubun) [inline liner.go]: That's right. Without this line, sometimes you get 2 braces.

Comment (motemen) [inline liner.go]: Ok, then shouldn't we use line erasing here? (like Ctrl-K)

Comment (k0kubun) [inline liner.go]: You're right. I changed it to use Ctrl-K and ported it for windows.  2203e97
I confirmed it works on windows again.

**Comment by motemen** [comment]
Thanks, I'll look into it shortly. Sorry for commenting code one by one, not at once.

**Comment by k0kubun** [comment]
By the way, I created a library to handle escape sequence, which also works on windows. It is [k0kubun/go-ansi](https://github.com/k0kubun/go-ansi). It includes cursor-up and erase-in-line. If you don't mind importing my library, I can remove the implementation for windows from this change. How do you think?

**Comment by motemen** [comment]
Finally I've read your code and merged with some modification!

Though I haven't read go-ansi the idea looks good, maybe we can think about it another place.

**Comment by k0kubun** [comment]
Thank you for merging and your fixes!

**Pull request merged**